arangod/Pregel/Conductor.cpp

```diff
@@ -319,6 +319,11 @@ VPackBuilder Conductor::finishedWorkerStep(VPackSlice const& data) {
           << "No further action taken after receiving all responses";
     }
   });
+  if (!queued) {
+    LOG_TOPIC("038db", ERR, Logger::PREGEL)
+        << "No thread available to queue response, canceling execution";
+    cancel();
```

Comment:
is `cancel()` going to call scheduler::queue again ?

Reply:
Hm... yes, you are right, via `_shutdownWorkers`.  I've modified the code there to make a retry attempt in case the queue is full. Could you comment on whether the retry logic is safe? I'm not 100% sure if something there can only happen once without generating further errors.